File: app/src/main/java/me/pwcong/jpstart/adapter/JPStartTabPagerAdapter.java
package me.pwcong.jpstart.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import java.util.List;

import me.pwcong.jpstart.component.fragment.JPStartFragment;
import me.pwcong.jpstart.mvp.bean.JPTab;

/**
 * Created by Pwcong on 2016/9/27.
 */

public class JPStartTabPagerAdapter extends BasePagerAdapter<JPTab> {

    public JPStartTabPagerAdapter(FragmentManager fm, List<JPTab> list) {
        super(fm, list);
    }

    @Override
    protected Fragment getFragment(JPTab jpTab) {
        return JPStartFragment.newInstance(jpTab.getType());
    }

    @Override
    protected CharSequence getTitle(JPTab jpTab) {
        return jpTab.getTitle();
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/adapter/JPStartTabPagerAdapter.kt
package com.github.pwcong.jpstart.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.github.pwcong.jpstart.mvp.bean.JPTab
import com.github.pwcong.jpstart.view.fragment.JPStartFragment

class JPStartTabPagerAdapter(fm: FragmentManager, list: List<JPTab>) :
    BasePagerAdapter<JPTab>(fm, list) {

    override fun getFragment(t: JPTab): Fragment {
        return JPStartFragment.newInstance(t.type)
    }

    override fun getTitle(t: JPTab): CharSequence? {
        return t.title
    }

}
